datafusion/core/src/physical_plan/aggregates/mod.rs

```diff
@@ -356,6 +409,23 @@ impl AggregateExec {
         )?;
 
         let schema = Arc::new(schema);
+        let mut aggregator_requirement = None;
+        // Ordering requirement makes sense only in Partial and Single modes.
+        // In other modes, all groups are collapsed, therefore their input schema
+        // can not contain expressions in the requirement.
+        if mode == AggregateMode::Partial || mode == AggregateMode::Single {
+            let requirement = get_finest_requirement(
+                &order_by_expr,
+                || input.equivalence_properties(),
+                || input.ordering_equivalence_properties(),
+            )?;
+            aggregator_requirement = requirement.map(|exprs| {
```

Comment:
I think this code could be simplified using https://docs.rs/datafusion-physical-expr/24.0.0/datafusion_physical_expr/struct.PhysicalSortRequirement.html#method.from_sort_exprs

Reply:
This is neat!. Thanks